CliClient/tests/filterParser.js

```diff
@@ -0,0 +1,131 @@
+	});
+
+	it('body with multiple words', () => {
+		const searchString = 'title:testTitle body:"word1 word2"';
+		const expected = new Map([
+			['title', [{ relation: 'AND', value: 'testTitle' }]],
+			['body', [{ relation: 'AND', value: 'word1' }, { relation: 'AND', value: 'word2' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('title or title', () => {
+		const searchString = 'title:testTitle1 OR title:testTitle2';
+		const expected = new Map([
+			['title', [{ relation: 'AND', value: 'testTitle1' }, { relation: 'OR', value: 'testTitle2' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('body or body', () => {
+		const searchString = 'body:testBody1 OR body:testBody2';
+		const expected = new Map([
+			['body', [{ relation: 'AND', value: 'testBody1' }, { relation: 'OR', value: 'testBody2' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('title or body', () => {
+		const searchString = 'title:testTitle1 OR body:testBody2';
+		const expected = new Map([
+			['title', [{ relation: 'AND', value: 'testTitle1' }]],
+			['body', [{ relation: 'OR', value: 'testBody2' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('single word text', () => {
+		const searchString = 'babayaga';
+		const expected = new Map([
+			['text', [{ relation: 'AND', value: 'babayaga' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('multi word text', () => {
+		const searchString = 'baba yaga';
+		const expected = new Map([
+			['text', [{ relation: 'AND', value: 'baba' }, { relation: 'AND', value: 'yaga' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('phrase text search', () => {
+		const searchString = '"baba yaga"';
+		const expected = new Map([
+			['text', [{ relation: 'AND', value: '"baba yaga"' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('tag', () => {
+		const searchString = 'tag:tag123';
+		const expected = new Map([
+			['tag', [{ relation: 'AND', value: 'tag123' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('multiple tags', () => {
+		const searchString = 'tag:tag123 tag:tag456';
+		const expected = new Map([
+			['tag', [{ relation: 'AND', value: 'tag123' }, { relation: 'AND', value: 'tag456' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('tag or tag', () => {
+		const searchString = 'tag:tag123 or tag:tag456';
+		const expected = new Map([
+			['tag', [{ relation: 'AND', value: 'tag123' }, { relation: 'OR', value: 'tag456' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+
+	it('multi word body', () => {
+		const searchString = 'body:"foo bar"';
+		const expected = new Map([
+			['body', [{ relation: 'AND', value: 'foo' }, { relation: 'AND', value: 'bar' }]],
+		]);
+		expect(filterParser(searchString)).toEqual(expected);
+	});
+});
```

Comment:
I think there should be tests to handle invalid filters too, such as `body:""foo"`, `titletitle:123`, `invalid:abc`, `:abc`, etc.

Reply:
Should we ignore the invalid filters and do the search anyway or throw an error?